pimcore/lib/Pimcore/Bundle/AdminBundle/Controller/Admin/AssetController.php

```diff
@@ -180,6 +182,14 @@ public function treeGetChildsByIdAction(Request $request)
             $childsList->setOrderKey("FIELD(assets.type, 'folder') DESC, assets.filename ASC", false);
 
             \Pimcore\Model\Element\Service::addTreeFilterJoins($cv, $childsList);
+
+            $beforeListLoadEvent = new GenericEvent($this, [
+                'list' => $childsList,
+                'context' => $allParams
+            ]);
+            \Pimcore::getEventDispatcher()->dispatch(AdminEvents::ASSET_LIST_BEFORE_LIST_LOAD, $beforeListLoadEvent);
```

Comment:
Maybe injecting directly event dispatcher could safe time in a future? There won't be need to refactor this.

Reply:
@Yivan can you please update your PR to use DI for the event dispatcher? Afterwards we're happy to merge your PR. 

Thanks!